libraries/chain/block_header_state.cpp

```diff
@@ -61,13 +62,29 @@ block_header_state block_header_state::next(block_header_state_input& input) con
    result.header = block_header {
       .timestamp         = input.timestamp, // [greg todo] do we have to do the slot++ stuff from the legacy version?
       .producer          = input.producer,
+      .confirmed         = hs_block_confirmed, // todo: consider 0 instead
       .previous          = input.parent_id,
       .transaction_mroot = input.transaction_mroot,
       .action_mroot      = input.action_mroot,
-      //.schedule_version = ?, [greg todo]
-      //.new_producers = ?     [greg todo]
+      .schedule_version  = header.schedule_version
    };
 
+   if(!proposer_policies.empty()) {
+      auto it = proposer_policies.begin();
+      if (it->first <= input.timestamp) {
+         result.active_proposer_policy = it->second;
+         result.header.schedule_version = header.schedule_version + 1;
+         result.active_proposer_policy->proposer_schedule.version = result.header.schedule_version;
+         result.proposer_policies = { ++it, proposer_policies.end() };
+      } else {
+         result.proposer_policies = proposer_policies;
```

Comment:
should we also have here:
`result.proposer_policy = proposer_policy;`

Reply:
I could have swore I had that in there. Yes, we want `result=active_proposer_policy = active_proposer_policy`. I can push a commit with that fix.

Reply:
pushed